extensions/typescript-language-features/src/tsServer/serverError.ts

```diff
@@ -82,12 +73,23 @@ export class TypeScriptServerError extends Error {
 	}
 
 	/**
-	 * Try to replace full TS Server paths with 'tsserver.js' so that we don't have to post process the data as much
+	 * Drop everything but ".js" and line/column numbers (though retain "tsserver" if that's the filename).
 	 */
-	private static normalizeMessageStack(version: TypeScriptVersion, message: string | undefined) {
+	private static sanitizeStack(message: string | undefined) {
 		if (!message) {
 			return '';
 		}
-		return message.replace(new RegExp(`${escapeRegExp(version.path)}[/\\\\]tsserver.js:`, 'gi'), 'tsserver.js:');
+		const regex = /(tsserver)?(\.(?:ts|tsx|js|jsx)(?::\d+(?::\d+))?)\)?$/igm;
```

Comment:
This moves us from an exclusion list to an inclusion list, so it should be impossible for customer data to slip through.